app/scripts/directives/serviceIntegration.js

```diff
@@ -67,12 +69,61 @@
 
     ctrl.getState = function(){
       if (ctrl.binding) {
-        return 1;
+        var state = 1;
+        if(ctrl.binding.status.conditions) {
+          _.each(ctrl.binding.status.conditions, (condition) => {
+            if(condition.type === "Ready" && condition.status !== "True"){
+              state = 2;
+            }
+          });
+        }
+        return state;
       }
       if (ctrl.serviceInstance){
         return 0;
       }
       return -1;
     };
+
+    ctrl.removeIntegration = function(){
+      var objectName = this.binding.metadata.name;
+      var context = {namespace: this.refApiObject.metadata.namespace};
+      var deleteOptions = {propagationPolicy: null};
+      DataService.delete({resource: 'podpresets', group: 'settings.k8s.io', version: 'v1alpha1'}, objectName, context, deleteOptions)
+      .then(function(){
+        DataService.get(APIService.getPreferredVersion('deployments'), _.get(ctrl, 'refApiObject.metadata.labels.serviceName'), context)
+        .then(function(deployment) {
+          //if there is a label in the deployment named after the integrated service, 
+          //delete the label and update the deployment to trigger a redeploy
+          if(ctrl.integrationData.name in deployment.spec.template.metadata.labels) {
+            delete(deployment.spec.template.metadata.labels[ctrl.integrationData.name]);
+            DataService.update(APIService.getPreferredVersion('deployments'), _.get(ctrl, 'refApiObject.metadata.labels.serviceName'), deployment, context)
+            .catch(function(error){
```

Comment:
I don't like to see nested promises. can you just return this promise and handle the notification based on the error in a single catch?